go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -118,9 +118,12 @@ def update_files # rubocop:disable Metrics/AbcSize
         def run_go_mod_tidy
           return unless tidy?
 
+          # NOTE(arslan): use `go mod tidy -e` once Go 1.16 is out:
+          # https://github.com/golang/go/commit/3aa09489ab3aa13a3ac78b1ff012b148ffffe367
           command = "go mod tidy"
+
           _, stderr, status = Open3.capture3(ENVIRONMENT, command)
-          handle_subprocess_error(stderr) unless status.success?
+          handle_subprocess_error(stderr, true) unless status.success?
```

Comment:
It seems to me like we're trying to silently ignore any problems that occur during `go mod tidy`. If that's accurate, then instead of adding a new arg to `handle_subprocess_error`, would it make sense to skip the invocation `handle_subprocess_error` entirely?

```suggestion
          # Attempt command, but ignore any errors
          Open3.capture3(ENVIRONMENT, command)
```

Reply:
This makes sense and I think we should use this idiom. Initially I changed this line and removed handling the error, but I wasn't sure about some of the edge cases and whether we should return the specific error types described above. I'll update the tests as well to accommodate the changes 👍🏼